File: ex05/ex05_06/ex05_06.py
TASK_LIST = [ 
    "wash clothes",
    "wash dishes",
    "clean house",
    "fix toilet",
    "cook dinner",
    "make coffee",
    "change light bulb",
    "water plants",
    "feed dog",
    "take out trash",
]

def main():
  task_option = 0
  index = 0
  tasks = [[-1] * 3 for _ in range(7)]

  for i in range(10):
    print(f"{i} - {TASK_LIST[i]}")

  for i in range(7):
    index = 0
    while True:
      print(f"Enter the task for day {i+1} (0-9): ")
      task_option = input()
      if not task_option.isdigit():
        tasks[i][index] = -1
        break
      task_option = int(task_option)
      if task_option < 0 or task_option > 9:
        print("Invalid task.")
        tasks[i][index] = -1
        break
      tasks[i][index] = task_option
      index += 1
      if index > 2:
        print("Next day tasks.")
        break
  for i in range(7):
    print(f"Day {i + 1} tasks: ")
    for j in range(3):
      if tasks[i][j] == -1:
        break
      print(f" {tasks[i][j]} - {TASK_LIST[tasks[i][j]]}")
    print(" ")

if __name__ == "__main__":
  main()
        

File: ex05/ex05_06/ex05_06.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <array>

const std::array<std::string, 10> TASK_LIST = {
    "wash clothes",
    "wash dishes",
    "clean house",
    "fix toilet",
    "cook dinner",
    "make coffee",
    "change bulb",
    "water plants",
    "feed dog",
    "take out trash",
};

int tasks[7][3];

int main() {
  int index = 0;
  char ch;
  int task_option = -1;
  for (int i = 0; i < 10; i++) {
    std::cout << i << " - " << TASK_LIST[i] << "\n";
  }
  for (int i = 0; i < 7; i++) {
    index = 0;
    do{
      std::cout << "Enter the task for day " << i+1 << " (0-9): ";
      ch = getchar();
      std::cout << "You entered: " << ch << "\n";
      while (getchar() != '\n'); 
      task_option = ch - '0';
      if (task_option < 0 || task_option > 9) {
        std::cout << "Invalid task.\n";
        tasks[i][index++] = -1;
        break;
      }
      tasks[i][index++] = task_option;
      if (index > 2) {
        std::cout << "Next day tasks." << "\n";
        break;
      }
    } while (true);
  }
  for (int i = 0; i < 7; i++) {
    std::cout << "Day " << i + 1 << " tasks: \n";
    for (int j = 0; j < 3; j++) {
      if (tasks[i][j] == -1) {
        break;
      }
      std::cout << " " << tasks[i][j] << " - " << TASK_LIST[tasks[i][j]] << "\n";
    }
    std::cout << "\n";
  }
  return 0;
}